pkg/aws/core.go

```diff
@@ -183,20 +183,15 @@ func (d *Driver) CreateMachine(ctx context.Context, req *driver.CreateMachineReq
 	// Set the AWS Capacity Reservation target. Using an 'open' preference means that if the reservation is not found, then
 	// instances are launched with regular on-demand capacity.
 	if providerSpec.CapacityReservationTarget != nil {
-		if providerSpec.CapacityReservationTarget.CapacityReservationResourceGroupArn != nil {
-			inputConfig.CapacityReservationSpecification = &ec2.CapacityReservationSpecification{
-				CapacityReservationPreference: aws.String("open"),
-				CapacityReservationTarget: &ec2.CapacityReservationTarget{
-					CapacityReservationResourceGroupArn: providerSpec.CapacityReservationTarget.CapacityReservationResourceGroupArn,
-				},
-			}
+		inputConfig.CapacityReservationSpecification = &ec2.CapacityReservationSpecification{}
```

Comment:
once all the validations have been done it is thus ensured that `providerSpec.CapacityReservationTarget` is properly set by the user. Then is there a need to have these if and else clauses? One can just directly initialise `inputConfig.CapacityReservationSpecification` right? If yes then just create a simple function to map.
```go
func mapToCapacityReservationSpecification(capacityReservationTarget *api.AWSProviderSpec.CapacityReservationTarget) *CapacityReservationSpecification {
// implementation
}
```
Then you can also write a separate unit test for it.

Reply:
Updated the code to just set the values now we know it's validated to provide only one value.